### PR TITLE
Add basic contributing guide

### PR DESCRIPTION
**Pull request by anishathalye**
Mainly added common-sense notes on submitting issues/PRs.

### CODE REVIEW

**Comment by anishathalye** [comment]
Related to this, it might be good to add a development guide (e.g. describing how to run the tests).

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,8 +1,55 @@
...
+
+Want to write code to help improve cleanlab? Awesome!
+
+If there are [open issues][issues], you're more than welcome to work on those.
+If you have your own ideas, that's great too! In that case, before working on
+substantial changes to the code base, it is *highly recommended* that you first
+[open an issue][issue] describing what you intend to work on.
+
+Any changes to the code base should try to follow the style and coding
+conventions used in the rest of the project.
```
Comment (jwmueller) [inline CONTRIBUTING.md:45]: Should we ask that:  All code should adhere to the [PEP8 style](https://www.python.org/dev/peps/pep-0008/)?

**Review by jwmueller — APPROVED** [review]
lgtm!

File: CONTRIBUTING.md (hunk 1 above)
Comment (anishathalye) [reply CONTRIBUTING.md:45]: I think standardizing code formatting is a great idea, but just asking people to follow a style is not enough. We should set up:

- An auto-formatter that's as easy to run as `go fmt` / `cargo fmt`, which will guarantee compliance with the style rules
- Checking in CI that code is formatted correctly

Python doesn't have a language-mandated standard for this, and I don't know off the top of my head which is the most popular community project for doing this (maybe [Black](https://black.readthedocs.io)?). I'll look into this and set it up as a separate PR.

**Pull request merged**